src/app/Library/CrudPanel/Traits/SaveActions.php

```diff
@@ -15,52 +19,260 @@ public function getSaveActionDefaultForCurrentOperation()
+            return $arr['order'] == $order;
+        });
+    }
+
+    /**
+     * Applys the registration of save actions.
+     *
+     * @param array $saveActions
+     * @return void
+     */
+    public function applyRegisterSaveActions($saveActions)
+    {
+        if (count($saveActions) != count($saveActions, COUNT_RECURSIVE)) {
+            foreach ($saveActions as $saveAction) {
+                $this->registerSaveAction($saveAction);
+            }
+        } else {
+            $this->registerSaveAction($saveActions);
+        }
+    }
+
+    /**
+     * Allow the developer to register save actions.
+     *
+     * @param array $saveActions
+     * @return void
+     */
+    public function registerSaveActions($saveActions)
+    {
+        if (is_array($saveActions)) {
+            $this->applyRegisterSaveActions($saveActions);
+        }
+    }
+
+    /**
+     * Register save actions in the crud.
+     *
+     * @param array $saveAction
+     * @return void
+     */
+    public function registerSaveAction($saveAction)
+    {
+        if (is_array($saveAction)) {
+            //check for some mandatory fields
+            $saveAction['name'] ?? abort(500, 'Please define save action name.');
+            $saveAction['redirect'] = $saveAction['redirect'] ?? function ($crud, $request, $itemId) {
+                return $request->has('http_referrer') ? $request->get('http_referrer') : $crud->route;
+            };
+            $saveAction['permissions'] = $saveAction['permissions'] ?? true;
+            $saveAction['button_text'] = $saveAction['button_text'] ?? $saveAction['name'];
+            $saveAction['order'] = isset($saveAction['order']) ? $this->reorderSaveActions($saveAction['order']) : count($this->availableSaveActions) + 1;
+            $this->availableSaveActions[$saveAction['name']] = [];
+            $this->availableSaveActions[$saveAction['name']] = $saveAction;
+        }
+    }
+
+    /**
+     * Replaces setting order or forces some default.
+     *
+     * @param int $wantedOrder
+     * @return int
+     */
+    public function reorderSaveActions($wantedOrder)
+    {
+        if (! empty($this->availableSaveActions)) {
+            $lastOrder = max(array_column($this->availableSaveActions, 'order'));
+            foreach ($this->availableSaveActions as &$sv) {
+                if ($wantedOrder == $sv['order']) {
+                    if (empty(array_filter($this->availableSaveActions, function ($arr) {
+                        $arr['order'] == 1;
+                    }))) {
+                        $sv['order'] = 1;
+                    } else {
+                        $sv['order'] = $lastOrder + 1;
+                    }
+                }
+            }
+        }
+
+        return $wantedOrder;
+    }
+
+    /**
+     * Replace the current save actions with the ones provided.
+     *
+     * @param array $saveActions
+     * @return void
+     */
+    public function replaceSaveActions($saveActions)
```

Comment:
A name like ```setSaveActions()``` would be more consistent with Fields, Columns, Filters, Buttons. 

**But!** 😄  I agree that ```replaceSaveActions()``` is a better name for it. Hands down 😄  It would have saved us some explanations having ```replaceColumns()``` and ```replaceFields()``` in the past.

So here's what I propose - let's use both. Let's keep the ```replaceSaveActions()``` method, but also add another method called ```setSaveActions()``` that just points to this one. An alias. That way we have the best of both worlds:
- a method with the better name - ```replaceSaveActions()```;
- a method with a consistent name with the other Features we have - ```setSaveActions()```;

Reply:
Just did it! :+1: